src/main/java/digital/slovensko/autogram/drivers/NativePkcs11SignatureToken.java

```diff
@@ -58,7 +58,7 @@ private void runContextSpecificLoginIfNeeded(Signature signature, PrivateKey pk)
             var p11 = getP11(signature);
             var sessionId = getSessionId(signature);
 
-            if (settings.getForceContextSpecificLoginEnabled() && isAlwaysAuthenticate(p11, sessionId, pk)) {
+            if (isAlwaysAuthenticate(p11, sessionId, pk) && (!isProtectedAuthenticationPath(p11, getSlotListIndex()) || settings.getForceContextSpecificLoginEnabled())) {
```

Comment:
```suggestion
            if (isAlwaysAuthenticate(p11, sessionId, pk) && (settings.getForceContextSpecificLoginEnabled() || !isProtectedAuthenticationPath(p11, getSlotListIndex()))) {
```

Ked to otocis tak to bude v pozitivnom pripade mierne rychlejsie lebo settings nerobi kartovu operaciu.